bottlecap/src/event_bus/bus.rs

```diff
@@ -0,0 +1,43 @@
+use std::sync::mpsc;
+use std::sync::mpsc::Sender;
+use std::thread;
+use tracing::error;
+
+use crate::events;
+use events::Event;
+
+pub struct EventBus {
+    tx: Option<Sender<events::Event>>,
+}
+
+impl EventBus {
+    pub fn new() -> EventBus {
+        EventBus { tx: None }
+    }
+    pub fn run(&mut self) {
+        let (tx, rx) = mpsc::channel::<Event>();
+        self.tx = Some(tx);
+        let _ = thread::spawn(move || loop {
```

Comment:
This drops the JoinHandle for the thread. We should probably keep that, so when we shut down we can guarantee this thread has finished processing.

Reply:
Another thought about the design. Perhaps it would be better if this run method was more like a "class method" or associated function, that does constructions of the EventBus in one step. That way an event bus object can always have a Sender, and you can remove the panic in get_sender_copy.

Reply:
yeah that's a good idea. I think I've addressed it in the last commit, let me know if this is what you had in mind